tools/system_libs.py

```diff
@@ -1480,6 +1480,13 @@ def calculate(temp_files, in_temp, cxx, forced, stdout_=None, stderr_=None):
   # TODO: Move all __deps from src/library*.js to deps_info.json, and use that single source of info
   #       both here and in the JS compiler.
   deps_info = json.loads(open(shared.path_from_root('src', 'deps_info.json')).read())
+
+  if not shared.Settings.WASM_BACKEND:
+    # fastcomp links differently than wasm-ld, and needs to be told explicitly
+    # about some dependencies
+    deps_info['setjmp'] += ['testSetjmp', 'saveSetjmp']
+    deps_info['longjmp'] += ['testSetjmp', 'saveSetjmp']
```

Comment:
So under the wasm backend what causes the linker to include testSetjmp and saveSetjmp?

Reply:
I think `wasm-ld` is just smart enough. it sees we need those symbols, so it keep looking in the `.a` files it has until it finds them. the fastcomp python linking code is not smart enough, and it just stops early, unless we tell it in advance "you will need these".

Reply:
I think I understand.. the references to testSetjmp and saveSetjmp in the fastcomp case are generated after bitcode linking... so they get DCE'd by the bitcode linker.  With wasm-ld the references are generated during codegen, before the linker runs, so the wasm-ld knows that it needs them.

Reply:
I do worry about LTO though, these symbols, like __THREW__ are generated by codegen which means they come into existence during LTO, and the current LTO pipeline still has some issues with this.

Reply:
Oh, in that case should they unconditionally be in `deps_info`, so they apply to upstream too?

Reply:
I don't see any harm in having them in there.. and it would make this code a little less ugly maybe?